docs/adr/01-admin-authentication.md

```diff
@@ -0,0 +1,62 @@
+# Authenticate to the admin interface
+
+## Status
+
+Proposed
+
+## Context
+
+The admin interface obviously cannot be public for the vast majority of websites. This means that we need to provide a way to handle authentication and authorization.
+
+### Characteristics
+
+We want this authentication system to be:
+
+- Easy to set up for nestjs-admin developpers (or better: working out-of-the-box)
+- Extensible to handle a wide range of use cases
+
+### What does django-admin do?
+
+Django-admin uses the builtin "user" model, that has a username and password. It also allows to
+
+### Challenges
+
+1. NestJS, as opposed to frameworks like Django, has no built-in concept of "user" with a username and password. We cannot rely on this.
+2. Users come in all shapes and form: identified through username, email, id, using a password or not... There's no invariant we can rely on.
+3. We need to make sure whatever solution we choose doesn't lock everybody out of the admin interface by default (we need to manually be able to create an admin, like django-admin with `./manage.py createsuperuser`)
+
+### Options
+
+1. Have the developper extend the Admin controller, which will allow them to define their own guard and authentication logic
+
+   - Very flexible
+   - Nothing to do for us but documentation
+   - Lot of friction to start using nestjs-admin, it makes the lib not usable out-of-the-box
+
+2. Add some config for the admin module to specify the User entity, which needs to implement some AdminUserInterface (getAdminUsername, getAdminPassword, isAdmin). We can then have a guard to implement authentication/authorization.
+
+   - Need to implement a login page
+   - Not very flexible, we'll have to make assumptions (user has a username and a password)
+
+3. Provide an AdminUser entity out-of-the-box, that has a username and password.
+
+   - Need to implement a login page
+   - Not very flexible. Admin entities are distinct from User entities, which is annoying (how to create admins? Choose their password?).
+   - Works entirely out-of-the-box
+
+## Decision
+
+The N.1 concern at the time is to make this library self-contained, usable out-of-the-box. For this reason, we'll go for option 1.
+
+The downsides of this option (flexibility, Users distinct from Admins) can be mitigated:
```

Comment:
Why not do no. 2 instead and allow users to extend the User entity for their own purpose a la django?

Reply:
Tempting, but it means the lib wouldn't work out-of-the-box, you need to touch your user model before (which might not exist if you're just starting a project)